W-14621089-W-14621092-ch2-transit-gateway-attachments (#170)
mulesoft/docs-hosting: cloudhub-2/modules/ROOT/pages/ps-create-configure-tgw.adoc

= Creating Transit Gateway Attachments

> When you attach a transit gateway to a private space, Amazon Web Services (AWS) creates a _transit gateway attachment_ to represent the connection. You can attach the same transit gateway to on-premisis VPCs in your own AWS account to enable communication between private spaces and VPCs. Multiple private spaces can connect to an internal VPC.

Reviewer: Cloudhub 2.0 creates the attachment on Amazon Web Services (AWS)

Author: ```suggestion
When you attach a transit gateway to a private space, CloudHub 2.0 creates a _transit gateway attachment_ to represent the connection on Amazon Web Services (AWS). You can attach the same transit gateway to on-premises VPCs in your own AWS account to enable communication between private spaces and VPCs. Multiple private spaces can connect to an internal VPC.
```

For download and usage information about the Network Tools application, see https://help.mulesoft.com/s/article/How-To-Use-Network-Tools-Application[How To Use Network Tools Application^] (KB article).

[[best-practices]]
== Maintain the Transit Gateway 

After you test the connection to your private space, periodically ensure your transit gateway connection is maintained. 

> If your private space loses access to the transit gateway, the state of the transit gateway is changed to `unknown` or `deleted`. If the transit gateway remains in the `unknown` state for 14 days, Anypoint Platform may remove the private space attachments from the transit gateway. However, the transit gateway itself is not removed. 

Reviewer: is the transit gateway remains in either `unknown` or `deleted` for 14 days, attachments can be removed

Author: ```suggestion
If your private space loses access to the transit gateway, the state of the transit gateway is changed to `unknown` or `deleted`. If the transit gateway remains in the `unknown` or `deleted` state for 14 days, Anypoint Platform may remove the private space attachments from the transit gateway. However, the transit gateway itself is not removed. 
```

* The transit gateway is deleted from the AWS console.
* A resource share is deleted.
* The MuleSoft account principal is deleted from the resource share. 

If you want to remove or delete the transit gateway from CloudHub 2.0, you should complete the steps in <<remove-tgw>> or <<delete-tgw>> before deleting the transit gateway, transit gateway attachment, or resource share from the AWS console. Removing connections from the AWS console without removing them from CloudHub 2.0 disrupts the connection but does not remove it. 

[[troubleshoot-tgw]]
== Troubleshoot Unknown or Deleted States for the Transit Gateway

When the transit gateway connection is disrupted, the state of the transit gateway changes to `unknown` or `deleted`. Depending on the cause of the state change, you can fix the state of the transit gateway by deleting or detaching the transit gateway.

The `unknown` state occurs when:

* The transit gateway or resource share is deleted. To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>.
* The MuleSoft account principal is deleted. To fix this state add the principal in the resource share.

> The `deleted` state occurs when the transit gateway is deleted from the AWS console but not CloudHub 2.0. To fix this state, detach the transit gateway from the private space using the steps in <<detach-tgw>>.

Author: ```suggestion
== Troubleshoot States for the Transit Gateway or Transit Gateway Attachment

When the transit gateway connection is disrupted, the state of the transit gateway or transit gateway attachment changes to `unknown`, `rejected`, or `deleted`. Depending on the cause of the state change, you can fix the state of the transit gateway or transit gateway attachment by deleting or detaching them.

=== Resolve Transit Gateway States

The `unknown` state for a transit gateway state occurs when:

* The transit gateway or resource share is deleted. To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>.
* The MuleSoft account principal is deleted. To fix this state, add the principal in the resource share.

The `deleted` state occurs when the transit gateway is deleted from the AWS console but not CloudHub 2.0. To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>.

=== Resolve Transit Gateway Attachment States

The `deleted` state for a transit gateway attachment occurs when the transit gateway attachment is deleted from the AWS console but not CloudHub 2.0. To fix this state, detach the transit gateway attachment from the private space using the instructions in <<detach-tgw>>.

The `rejected` state for a transit gateway attachment occurs when the transit gateway attachment is rejected. To fix this state, detach the transit gateway attachment from the private space using the steps in <<detach-tgw>>.

```

For download and usage information about the Network Tools application, see https://help.mulesoft.com/s/article/How-To-Use-Network-Tools-Application[How To Use Network Tools Application^] (KB article).

[[best-practices]]
== Maintain the Transit Gateway 

> After you test the connection to your private space, periodically ensure your transit gateway connection is maintained. 

Reviewer 2: ```suggestion
After you test the connection to your private space, periodically maintain your transit gateway connection. 
```

Reviewer 2: What does maintaining include?

Reviewer 2: Should maintain be "check the connection to"?

Author: ```suggestion
After you test the connection to your private space, periodically check the connection to your transit gateway. 
```

> If your private space loses access to the transit gateway, the state of the transit gateway is changed to `unknown` or `deleted`. If the transit gateway remains in the `unknown` or `deleted` state for 14 days, Anypoint Platform may remove the private space attachments from the transit gateway. However, the transit gateway itself is not removed. 

Reviewer 2: ```suggestion
If your private space loses access to the transit gateway, the state of the transit gateway changes to `unknown` or `deleted`. If the transit gateway remains in the `unknown` or `deleted` state for 14 days, Anypoint Platform might remove the private space attachments from the transit gateway. However, the transit gateway itself is not removed. 
```

Your private space loses access to a transit gateway when:

* The transit gateway is deleted from CloudHub 2.0.
* The transit gateway is deleted from the AWS console.
* A resource share is deleted.
* The MuleSoft account principal is deleted from the resource share. 

> If you want to remove or delete the transit gateway from CloudHub 2.0, you should complete the steps in <<remove-tgw>> or <<delete-tgw>> before deleting the transit gateway, transit gateway attachment, or resource share from the AWS console. Removing connections from the AWS console without removing them from CloudHub 2.0 disrupts the connection but does not remove it. 

Reviewer 2: ```suggestion
To remove or delete the transit gateway from CloudHub 2.0, complete the steps in <<remove-tgw>> or <<delete-tgw>> before deleting the transit gateway, transit gateway attachment, or resource share from the AWS console. Removing connections from the AWS console without removing them from CloudHub 2.0 disrupts the connection but does not remove it. 
```

[[troubleshoot-tgw]]
== Troubleshoot States for the Transit Gateway or Transit Gateway Attachment

> When the transit gateway connection is disrupted, the state of the transit gateway or transit gateway attachment changes to `unknown`, `rejected`, or `deleted`. Depending on the cause of the state change, you can fix the state of the transit gateway or transit gateway attachment by deleting or detaching them.

Reviewer 2: Would this fix also include re-adding or reattaching?

Reviewer 2: Or adding the principal?

Author: I will check.

Author: ```suggestion
When the transit gateway connection is disrupted, the state of the transit gateway or transit gateway attachment changes to `unknown`, `rejected`, or `deleted`. Depending on the cause of the state change, you can fix the state of the transit gateway or transit gateway attachment by deleting and recreating or detaching and reattaching them.
```

=== Resolve Transit Gateway States

The `unknown` state for a transit gateway state occurs when:

> * The transit gateway or resource share is deleted. To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>.

Reviewer 2: ```suggestion
* The transit gateway or resource share is deleted.
+
To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>.
```

> * The MuleSoft account principal is deleted. To fix this state, add the principal in the resource share.

Reviewer 2: ```suggestion
* The MuleSoft account principal is deleted.
+
To fix this state, add the principal in the resource share.
```

To remove or delete the transit gateway from CloudHub 2.0, complete the steps in <<remove-tgw>> or <<delete-tgw>> before deleting the transit gateway, transit gateway attachment, or resource share from the AWS console. Removing connections from the AWS console without removing them from CloudHub 2.0 disrupts the connection but does not remove it. 

[[troubleshoot-tgw]]
== Troubleshoot States for the Transit Gateway or Transit Gateway Attachment

When the transit gateway connection is disrupted, the state of the transit gateway or transit gateway attachment changes to `unknown`, `rejected`, or `deleted`. Depending on the cause of the state change, you can fix the state of the transit gateway or transit gateway attachment by deleting and recreating or detaching and reattaching them.

=== Resolve Transit Gateway States

The `unknown` state for a transit gateway state occurs when:

* The transit gateway or resource share is deleted.
+
To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>.
* The MuleSoft account principal is deleted.
+
To fix this state, add the principal in the resource share.

> The `deleted` state occurs when the transit gateway is deleted from the AWS console but not CloudHub 2.0. To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>.

Author: ```suggestion
The `deleted` state occurs when the transit gateway is deleted from the AWS console but not CloudHub 2.0. To fix this state, delete the transit gateway from CloudHub 2.0 using the steps in <<delete-tgw>>, then create a new transit gateway.
```

=== Resolve Transit Gateway Attachment States

> The `deleted` state for a transit gateway attachment occurs when the transit gateway attachment is deleted from the AWS console but not CloudHub 2.0. To fix this state, detach the transit gateway attachment from the private space using the instructions in <<detach-tgw>>.

Author: ```suggestion
The `deleted` state for a transit gateway attachment occurs when the transit gateway attachment is deleted from the AWS console but not CloudHub 2.0. To fix this state, detach the transit gateway attachment from the private space using the steps in <<detach-tgw>>, then reattach the transit gateway attachment.
```

> The `rejected` state for a transit gateway attachment occurs when the transit gateway attachment is rejected. To fix this state, detach the transit gateway attachment from the private space using the steps in <<detach-tgw>>.

Author: ```suggestion
The `rejected` state for a transit gateway attachment occurs when the transit gateway attachment is rejected. To fix this state, detach the transit gateway attachment from the private space using the steps in <<detach-tgw>>, then reattach the transit gateway attachment.
```
